internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -90,6 +95,30 @@ func (i *Infra) createOrUpdateDeployment(ctx context.Context, r ResourceRender)
 	})
 }
 
+// createOrUpdateDaemonSet creates a DaemonSet in the kube api server based on the provided
+// ResourceRender, if it doesn't exist and updates it if it does.
+func (i *Infra) createOrUpdateDaemonSet(ctx context.Context, r ResourceRender) error {
+	daemonSet, err := r.DaemonSet()
+	if err != nil {
+		return err
+	}
+
+	// skip daemonset creation if it is not needed
+	if daemonSet == nil {
```

Comment:
Same as the comment on the deployment deletion

Reply:
Yes. What's wrong?